### PR TITLE
sensor: icm42688 configuration and regmap fixes

### PR DESCRIPTION
**Pull request by teburd**
The int config and fifo config register addresses were wrong, fix those.

Adds lots of debug information (when LOG_DBG=y) to the configuration of the device which is incredibly helpful for diagnosing configuration issues.

This fixes data ready interrupts with the trigger support @yperess is working on.

Signed-off-by: Tom Burdick <thomas.burdick@intel.com>

### CODE REVIEW

**Comment by laurenmurphyx64** [comment]
Hi @teburd, if you need this in for 3.3, file and link a bug - if not carry on

**Comment by teburd** [comment]
This does more than fix a bug, so it can wait for 3.4

**Review by yperess — DISMISSED** [review]

File: drivers/sensor/icm42688/icm42688_common.c [hunk 1]
```diff
@@ -69,6 +69,9 @@ int icm42688_configure(const struct device *dev, struct icm42688_cfg *cfg)
 	const struct icm42688_dev_cfg *dev_cfg = dev->config;
 	int res;
 
+	/* Disable interrupts, reconfigured at end */
+	res = icm42688_spi_single_write(&dev_cfg->spi, REG_INT_SOURCE0, 0);
```
Comment (MaureenHelm) [inline drivers/sensor/icm42688/icm42688_common.c:73]: Adding new SPI transactions looks suspicious when the commit message says you're fixing register addresses and adding log messages. Can you update the commit message to explicitly call out that you're fixing data ready interrupts? You mention it in the PR summary but would be helpful to have that information in the commit history.

**Review by MaureenHelm — APPROVED** [review]

**Comment by carlescufi** [comment]
@yperess please take a look

**Review by yperess — APPROVED** [review]

**Pull request merged**